consensus/src/liveness/proposal_generator.rs

```diff
@@ -139,6 +147,46 @@ impl PipelineBackpressureConfig {
                 v
             })
     }
+
+    pub fn get_execution_block_size_backoff(
+        &self,
+        block_execution_times: &[ExecutionSummary],
+    ) -> Option<u64> {
+        info!(
+            "Estimated block execution times: {:?}",
+            block_execution_times
+        );
+
+        self.execution.as_ref().and_then(|config| {
+            let sizes = block_execution_times
+                .iter()
+                .flat_map(|summary| {
+                    let execution_time_ms = summary.execution_time.as_millis();
+                    if execution_time_ms > config.min_block_time_ms_to_activate as u128 {
```

Comment:
I am wondering if this is the optimal way to calculate the block size. Instead of this, how about calculating some sort of normalize time to execute per transaction in the entire window and use that to compute the block size?

Reply:
updated comment why I am doing it this way